packages/datepicker/src/js/index.ts

```diff
@@ -0,0 +1,136 @@
+import { DateParts } from '../react'
+import { MonthDateParts } from '../react/calendar'
+
+const num = (n: string) => parseInt(n, 10)
+
+export const arrayOf = (num: string): unknown[] =>
+  Array.apply(null, Array(parseInt(num, 10) || 0))
+
+export const getDaysInMonth = ({ mm, yyyy }: MonthDateParts): string => {
+  const lastDayInPreviousMonth = 0
+  return new Date(parseInt(yyyy, 10), parseInt(mm, 10), lastDayInPreviousMonth)
+    .getDate()
+    .toString()
+}
+
+export const forceValidDay = (date?: DateParts): string => {
+  if (!date) return ''
+
+  const maxDays = parseInt(getDaysInMonth(date), 10)
+  const day = parseInt(date.dd, 10)
+  return isNaN(day)
+    ? ''
+    : day > maxDays
+    ? maxDays.toString()
+    : day < 1
+    ? '1'
+    : day.toString()
```

Comment:
not new...but wanted to complain that this is hard to read 😛

Reply:
I will hear your complaint and add another TS-approved if/else statement.